common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -46,6 +47,42 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
    * @return true if the arrays are equal, false otherwise
    */
   public static boolean arrayEquals(
+      MemoryBlock leftBase, long leftOffset, MemoryBlock rightBase, long rightOffset, final long length) {
```

Comment:
Is there any semantic difference between this method and one right below ? Can we do `getBaseObject()` over the `leftBase` and `rightBase` and call the method below at L#85 ?

Reply:
sure, done.

Reply:
Not sure if you missed this. I still see the code duplication. This entire method block could be replaced as:

```
arrayEquals(
  leftBase.getBaseObject(),
  leftOffset,
  rightBase.getBaseObject(),
  rightOffset,
  length
)
```

Reply:
Thanks, I missed this.